scylladb/scylladb
# docs: add swagger ui extension

Depends on https://github.com/scylladb/scylladb/pull/18252

Closes https://github.com/scylladb/scylladb/issues/8180

## Motivation

Renders the API Reference from [api/api-doc](https://github.com/scylladb/scylladb/tree/master/api/api-doc) using [Swagger UI 2.2.10](https://github.com/swagger-api/swagger-ui?tab=readme-ov-file#compatibility).

## Limitations

- **Schemas from external files:** Some of the "Storage proxy" endpoints define schemas that are loaded from a external file named `utils.json` ([example](https://github.com/scylladb/scylladb/blob/master/api/api-doc/storage_proxy.json#L725)). This version of Swagger UI does not support including schemas from different files, which results in these schemas not being rendered.

- **Navigation highlighting issues:** The right-side navigation highlighting does not behave as expected. When a heading is clicked, sometimes it incorrectly highlights the previous section instead of the current one.  Will be fixed by https://github.com/scylladb/sphinx-scylladb-theme/pull/1092

## How to test

1. Clone this PR.
2. Build the docs:

    ```
    cd docs
    make preview
    ```
3. Open http://127.0.0.1:5500/reference/api-reference

    You should now view the API reference documentation.
